Examples/SnapinsSDKKotlinExample/app/src/javaImplDebug/java/com/salesforce/snapinssdkexample/SupportHomeViewAddition.java

```diff
@@ -0,0 +1,236 @@
+    }
+
+    /**
+     * Configures and launches Live Agent Chat
+     */
+    private void launchChat() {
+        // Create a UI configuration instance from a core config object
+        ChatConfiguration chatConfiguration = ServiceSDKUtils.getChatConfigurationBuilder(context)
+                .preChatFields(buildPreChatFields())
+                .build();
+
+        // Configure chat session listener
+        ServiceSDKApplication serviceSDKApplication = (ServiceSDKApplication) context.getApplicationContext();
+        final ChatSessionListener chatListener = serviceSDKApplication.getChatSessionListener();
+
+        ChatUI.configure(ChatUIConfiguration.create(chatConfiguration))
+                .createClient(context)
+                .onResult(new Async.ResultHandler<ChatUIClient>() {
+                    @Override
+                    public void handleResult(Async<?> async, @NonNull ChatUIClient chatUIClient) {
+                            // Add the configured chat session listener to the Chat UI client
+                            chatUIClient.addSessionStateListener(chatListener);
+                            // Start the live agent chat session
+                            chatUIClient.startChatSession((FragmentActivity) context);
+                    }
+                });
+    }
+
+    /**
+     * Configures pre chat fields if prechat is enabled in settings
+     */
+    private List<PreChatField> buildPreChatFields(){
+        PreChatField preChatField1 = new PreChatField.Builder()
+                    .build(context.getString(R.string.prechat_agent_info_label),
+                            context.getString(R.string.prechat_enter_name_label),
+                            PreChatField.STRING);
+
+        PreChatField preChatField2 = new PreChatField.Builder()
+                .required(true)
+                .addPickListOption(new PreChatField.PickListOption(
+                        context.getString(R.string.prechat_example_id) + 1,
+                        context.getString(R.string.prechat_example_selection_one)))
+                .addPickListOption(new PreChatField.PickListOption(
+                        context.getString(R.string.prechat_example_id) + 2,
+                        context.getString(R.string.prechat_example_selection_two)))
+                .build(context.getString(R.string.prechat_agent_info_label),
+                        context.getString(R.string.prechat_selection_label_title),
+                        PreChatField.PICKLIST);
+
+        return preChatEnabled()
+                ? Utils.asMutableList(preChatField1, preChatField2)
+                : Collections.<PreChatField>emptyList();
+    }
+
+    /**
+     * Helper method to determine if pre chat is enabled
+     */
+    private boolean preChatEnabled() {
+        return getBooleanPref(
+                context, ChatSettingsActivity.KEY_PRECHAT_ENABLED);
+    }
+
+    /**
+     * Configures and launches SOS
+     */
+    private Boolean launchSos() {
```

Comment:
I avoid using non primitive data types whenever possible (see https://docs.oracle.com/javase/tutorial/java/nutsandbolts/datatypes.html) (use the lowercase `boolean` instead of `Boolean`). I suggest this because `Boolean` can be null (`boolean` can't be null) and potentially cause a `NullPointerException`.

Reply:
done